bin/ls.py

```diff
@@ -77,96 +73,132 @@ def is_image(path):
         return True
     else:
         return False
-
+        
+def human_legible_size(num):
+    """
+    Return a human readable string describing the size of something.
+    :param num: the number in machine-readble form
+    :type num: int
+    """
+    suffix = ''
+    if num<=999999:
+        return '%7d' % num
+    for suffix in ['K','M','G','T']:
+        num /= 1024.0
+        if num<=9999:
+            return '%4.1f%s' % (num,suffix)
+    return "%4.1f%s" % (num, 'P')
+    
+def unformatted_size(num):
+    return '%7d' % num
+    
+def sortKey(txt):
+    result = ''
+    for c in txt:
+        if c.islower():
+            result += c+'1'
+        if c.isupper():
+            result += c.lower()+'2'
+        else:
+            result += c+'0'
+    return result
+    
 def main(args):
-
-    ap = ArgumentParser()
+    ap = ArgumentParser(add_help=False)
+    ap.add_argument('--help', action='help', help='show this help message and exit')
```

Comment:
I'm curious: Is there a specific reason you manually add `--help`?

Reply:
I introduced the option -h for human-legible size and that is usually interpreted as an alias for --help by ArgumentParser. Except if you use add_help=False. But then --help doesn't work anymore either, except if you explicitly add it to the arguments - what I did here.

Reply:
Ah, I see. Thank you for the explanation.